front/app/components/AddToBasketButton/useAssignBudget.ts

```diff
@@ -0,0 +1,157 @@
+import { useCallback, useState } from 'react';
+
+// hooks
+import useAuthUser from 'api/me/useAuthUser';
+import useIdeaById from 'api/ideas/useIdeaById';
+import useProjectById from 'api/projects/useProjectById';
+import usePhases from 'api/phases/usePhases';
+import useBasket from 'api/baskets/useBasket';
+import useUpdateBasket from 'api/baskets/useUpdateBasket';
+import useAddBasket from 'api/baskets/useAddBasket';
+
+// utils
+import { getParticipationContext } from './utils';
+import { capitalizeParticipationContextType } from 'utils/helperUtils';
+import eventEmitter from 'utils/eventEmitter';
+
+// tracks
+import { trackEventByName } from 'utils/analytics';
+import tracks from 'components/AddToBasketButton/tracks';
+
+// constants
+import { BUDGET_EXCEEDED_ERROR_EVENT } from 'components/ParticipationCTABars/VotingCTABar/events';
+
+// typings
+import { BasketIdeaAttributes } from 'api/baskets/types';
+
+const timeout = (ms: number) =>
+  new Promise((resolve) => setTimeout(resolve, ms));
+
+interface Props {
+  projectId: string;
+  ideaId: string;
+}
+
+const useAssignBudget = ({ projectId, ideaId }: Props) => {
+  const [processing, setProcessing] = useState(false);
+  const { data: authUser } = useAuthUser();
+  const { data: idea } = useIdeaById(ideaId);
+  const { data: project } = useProjectById(projectId);
+  const { data: phases } = usePhases(projectId);
+  const { mutateAsync: addBasket } = useAddBasket(projectId);
+  const { mutateAsync: updateBasket } = useUpdateBasket();
+
+  const participationContext = getParticipationContext(project, idea, phases);
+
+  const { data: basket } = useBasket(
+    participationContext?.relationships?.user_basket?.data?.id
+  );
+
+  const assignBudget = useCallback(async () => {
+    if (!authUser || !idea || !participationContext) {
+      return;
+    }
+
+    const participationContextType =
+      project?.data.attributes.process_type === 'continuous'
+        ? 'project'
+        : 'phase';
+
+    const participationContextId = participationContext.id;
+    const maxBudget = participationContext?.attributes.voting_max_total;
+    const ideaBudget = idea?.data.attributes.budget;
+    const basketTotal = basket?.data.attributes.total_votes;
+
+    const done = async () => {
+      await timeout(200);
```

Comment:
Why do we need this timeout? Could we use the "Loading" state from useUpdateBasket instead? :thinking:

Reply:
Not sure, I just copied what was already there... I will try to remove it